aten/src/ATen/native/Resize.cpp

```diff
@@ -18,10 +18,22 @@ void resize_output(Tensor& output, IntArrayRef shape) {
       "reuse an out tensor t by resizing it, inplace, to zero elements with ",
       "t.resize_(0).");
   }
+}
 
+void resize_output(Tensor& output, IntArrayRef shape) {
+  resize_output_check(output, shape);
   output.resize_(shape);
```

Comment:
Hey, do you think it makes sense to put
```
if (output.is_cpu()){
    at::native::resize_(output_shape);
 else {
   output.resize_(shape);
 }
```
here instead of TI and codegen? That way, all callers of resize_output get the benefit.

Reply:
oh yeah, that seems much cleaner (at the cost of one extra tiny branch). I'll add that to my list of stuff to clean up :)

Reply:
But you are getting rid of that branch in TI. It's true that structured kernels will have extra branch in their hot path though :-( but I still think it's worth it.